src/plugins/data_source/common/data_sources/types.ts

```diff
@@ -11,8 +11,15 @@ export interface DataSourceAttributes extends SavedObjectAttributes {
   endpoint: string;
   auth: {
     type: AuthType;
-    credentials: UsernamePasswordTypedContent | undefined;
+    credentials: UsernamePasswordTypedContent | SigV4Content | undefined;
   };
+  lastUpdatedTime?: string;
+}
+
+export interface SigV4Content extends SavedObjectAttributes {
```

Comment:
why does `SigV4Content` extends `SavedObjectAttributes`?

Reply:
removed savedObjectAttr on  both username&password  and sigv4 interfaces